packages/dev/core/src/Materials/greasedLinePluginMaterial.ts

```diff
@@ -0,0 +1,551 @@
+                    vec4 nextPos = m * vec4( next, 1.0 );
+
+                    vec2 currentP = fix( finalPosition, aspect );
+                    vec2 prevP = fix( prevPos, aspect );
+                    vec2 nextP = fix( nextPos, aspect );
+
+                    float w =  lineWidth * widths;
+
+                    vec2 dir;
+                    if( nextP == currentP ) dir = normalize( currentP - prevP );
+                    else if( prevP == currentP ) dir = normalize( nextP - currentP );
+                    else {
+                    vec2 dir1 = normalize( currentP - prevP );
+                    vec2 dir2 = normalize( nextP - currentP );
+                    dir = normalize( dir1 + dir2 );
+                    }
+                    vec4 normal = vec4( -dir.y, dir.x, 0., 1. );
+                    normal.xy *= .5 * w;
+                    normal *= greasedLineProjection;
+                    #ifdef GREASED_LINE_SIZE_ATTENUATION
+                    normal.xy *= finalPosition.w;
+                    normal.xy /= ( vec4( resolution, 0., 1. ) * greasedLineProjection ).xy;
+                    #endif
+
+                    finalPosition.xy += normal.xy * side;
+
+                    gl_Position = finalPosition;
+
+                    vNormal= normal.xyz;
+                    vPositionW = vec3(finalPosition);
+                `,
+                // eslint-disable-next-line @typescript-eslint/naming-convention
+                "!gl_Position\\=viewProjection\\*worldPos;": "//", // remove
+            };
+        }
+
+        if (shaderType === "fragment") {
+            return {
+                // eslint-disable-next-line @typescript-eslint/naming-convention
+                CUSTOM_FRAGMENT_DEFINITIONS: `
+                    varying float vCounters;
+                    flat in int vColorPointers;
+                `,
+                // eslint-disable-next-line @typescript-eslint/naming-convention
+                CUSTOM_FRAGMENT_MAIN_END: `
+                    gl_FragColor.a *= step(vCounters, greasedLineVisibility);
+                    if( gl_FragColor.a == 0. ) discard;
+
+                    if(useDash == 1.){
+                        gl_FragColor.a *= ceil(mod(vCounters + dashOffset, dashArray) - (dashArray * dashRatio));
+                        if(gl_FragColor.a == 0.) discard;
+                    }
+
+                    #ifdef GREASED_LINE_HAS_COLOR
+                        if (colorMode == ${GreasedLineMeshColorMode.COLOR_MODE_SET}.) {
+                        gl_FragColor.rgb = singleColor;
+                        } else if (colorMode == ${GreasedLineMeshColorMode.COLOR_MODE_ADD}.) {
+                        gl_FragColor.rgb += singleColor;
+                        } else if (colorMode == ${GreasedLineMeshColorMode.COLOR_MODE_MULTIPLY}.) {
+                        gl_FragColor.rgb *= singleColor;
+                        }
+                    #else
+                        if (useColors == 1.) {

```

Comment:
I would like to have fast switching of colors using the useColors option and to avoid shader recompilation I added a dummy 4 byte long texture when in WebGPU which is set if there is no colors options (thus no texture) specified. Is it OK?